### DISCUSSION TITLE
What's the rationale to have "bring" expressions instead of "import" expressions?

### DISCUSSION THREAD

**Discussion by gmeligio** [winglang/wing#779] [Q&A] [answered]
My questions are regarding readability for developers coming from other languages, related to this [design tenet
](https://docs.winglang.io/reference/spec#02-design-tenets)
> Developers coming from other mainstream cloud languages (C#, Java, and TS) should feel right at home.

I looked at the issues and discussions to see if there was information related to my questions but didn't find any.

When looking at the example on the [home page](https://www.winglang.io/), the first statement has the `bring` keyword, which led me to think it meant something different than import. Later looking at the spec, I found in [2.1 bring](https://docs.winglang.io/reference/spec#21-bring)

> "bring" statement can be used to import and reuse code

and then in the details about the [module system](https://docs.winglang.io/reference/spec#41-imports) it says `bring` is used

> To import a JSII / Wing package under a named import

In both places, the word to express the semantic of `bring` is import. So for me, it looks like `bring` intends roughly the same as `import` in other languages like TypeScript, Python, Go and Java. At first, I thought that the use of `bring` was because the behaviour is different from `import`, but it seems they are pretty similar. Another possibility is that `import` is reserved for other use cases. I think `import` is more readable, even if some features/improvements are still missing.

## Questions

1. Is there a reason it's called `bring` instead of `import`?
1. Is `import` planned for other/future use cases?

**Comment by 3p3r** [comment, marked as answer]
Choice of keyword here is deliberate. An import/include/bring statement is used to fetch code that’s elsewhere and make it available here.

My theory is that, the bring statement is often the very first thing you’d notice in a source code online, and if it’s alien enough, you are more likely to look it up and go “hey what language is that?”.

Which by the existence of this ticket, it proves my theory was right :)

Aside from that, I have design plans for bring’s future. Been experimenting with the concept of bundling raw source code as JSII modules during compile time.

**Reply by gmeligio** [reply]
Thank you for your response. I got curious about the language right after looking at the short snippet as a whole together with the animated diagram. In my case, `bring` only causes me confusion, it made me wonder if the other keywords meant what I expected from other languages and then read the spec. I like the overall features and syntax but because I work with Python and TypeScript these questions came to my mind. I really look forward to the JSII bundling and wing future.
